File: backend/slotify/spring/src/main/java/com/slotify/backend/spring/company/services/CompanyService.java
package com.slotify.backend.spring.company.services;

import com.slotify.backend.spring.company.enums.CompanyFetchMode;
import com.slotify.backend.spring.company.models.CompanyEntity;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

public interface CompanyService {

    CompanyEntity saveCompany(CompanyEntity companyEntity);

    Optional<CompanyEntity> findCompanyById(UUID companyId);

    List<CompanyEntity> findCompaniesById(List<UUID> companiesId);

    Optional<CompanyEntity> findCompanyByIdAndFetchMode(UUID companyId, CompanyFetchMode fetchMode);
}

File: frontend/app/src/main/java/com/raj/slotify/services/retrofit/slotifyBackend/CompanyService.kt
package com.raj.slotify.services.retrofit.slotifyBackend

import com.raj.slotify.dtos.company.GetCompanyResponse
import com.raj.slotify.dtos.company.GetServicesResponse
import com.raj.slotify.dtos.company.PatchCompanyResponse
import com.raj.slotify.dtos.company.SearchCompaniesResponse
import com.raj.slotify.dtos.user.UserSummary
import com.raj.slotify.models.api.PageResponse
import okhttp3.MultipartBody
import okhttp3.RequestBody
import retrofit2.Response
import retrofit2.http.*
import java.util.UUID

interface CompanyService {

    @GET("companies/search")
    suspend fun getCompaniesByQuery(
        @Header("Authorization") authHeader: String,
        @Query("q") q: String,
        @Query("page") page: Int,
    ): Response<PageResponse<SearchCompaniesResponse>>

    @Multipart
    @PATCH("companies")
    suspend fun patchCompany(
        @Header("Authorization") authHeader: String,
        @Part file: MultipartBody.Part?,
        @Part("request") request: RequestBody
    ): Response<PatchCompanyResponse>

    @GET("companies/{companyId}")
    suspend fun getCompanyById(
        @Path("companyId") companyId: UUID,
        @Header("Authorization") authHeader: String,
        @Query("fetchMode") fetchMode: String?,
    ): Response<GetCompanyResponse>

    @GET("companies/{companyId}/services")
    suspend fun getServicesByCompanyId(
        @Path("companyId") companyId: UUID,
        @Header("Authorization") authHeader: String,
        @Query("fetchMode") fetchMode: String?,
        @Query("page") page: Int?,
        @Query("sortBy") sortBy: String?,
        @Query("order") order: String?
    ): Response<PageResponse<GetServicesResponse>>

}